src/service/plant_type.ts

```diff
@@ -0,0 +1,351 @@
+      if (parserResult.success) {
+        const plantTypeUuid: string = uuidv4();
+        const plantTypeDatabase: PlantTypeDatabase = {
+          PK: `PLANT_TYPE#${plantTypeUuid}`,
+          SK: plantTypeUuid,
+          type: "PLANT_TYPE",
+          GSI: parserResult.data.name,
+          GSI2: "",
+          data: {
+            name: parserResult.data.name,
+            picture: parserResult.data.picture,
+            waterRequirement: parserResult.data.waterRequirement,
+            sunRequirement: parserResult.data.sunRequirement,
+          },
+        };
+        const createPlantTypeCommand = async () =>
+          await db.send(
+            new PutCommand({
+              TableName: TABLE_NAME,
+              Item: plantTypeDatabase,
+            }),
+          );
+        const createPlantTypeResult = await processRequest(
+          createPlantTypeCommand,
+          "createPlantType",
+        );
+        if (createPlantTypeResult.success) {
+          return createRequestSuccess("createPlantType")(
+            parserResult.data,
+            createPlantTypeResult.code,
+            createPlantTypeResult.message,
+          );
+        } else {
+          return createPlantTypeResult;
+        }
+      } else {
+        return parserResult;
+      }
+    },
+    async updatePlantType(
+      req: UpdatePlantTypeRequest,
+    ): Promise<RequestResult<"updatePlantType", PlantType>> {
+      const getPlantTypeCommand = async () => {
+        const { Item } = await db.send(
+          new GetCommand({
+            TableName: TABLE_NAME,
+            Key: {
+              PK: `PLANT_TYPE#${req.payload.uuid}`,
+              SK: req.payload.uuid,
+            },
+          }),
+        );
+        return Item;
+      };
+      const getPlantTypeResult = await processRequest(
+        getPlantTypeCommand,
+        "updatePlantType",
+      );
+      if (!getPlantTypeResult.success) {
+        return getPlantTypeResult;
+      }
+      const item = req.payload;
+      const parserResult = parseData(item, "updatePlantType", PlantTypeSchema);
+      if (parserResult.success) {
+        const plantTypeDatabase: PlantTypeDatabase = {
+          PK: `PLANT_TYPE#${parserResult.data.SK}`,
+          SK: parserResult.data.SK,
+          type: "PLANT_TYPE",
+          GSI: req.payload.name,
+          GSI2: "",
+          data: {
+            name: req.payload.name,
+            picture: req.payload.picture,
+            waterRequirement: req.payload.waterRequirement,
+            sunRequirement: req.payload.sunRequirement,
+          },
+        };
+        const updatePlantTypeCommand = async () =>
+          await db.send(
+            new PutCommand({
+              TableName: TABLE_NAME,
+              Item: plantTypeDatabase,
+            }),
+          );
+        const updatePlantTypeResult = await processRequest(
+          updatePlantTypeCommand,
+          "updatePlantType",
+        );
+        if (updatePlantTypeResult.success) {
```

Comment:
Error first approach. No more ELSE, if something is not what you want it to be, return.